src/vaadin-checkbox.html

```diff
@@ -241,7 +241,7 @@
         }
 
         _handleClick(e) {
-          if (!this.disabled) {
+          if (!this.disabled && !(e.target.matches || e.target.matchesSelector).call(e.target, 'a')) {
```

Comment:
1. It's better to check `e.target.localName !== 'a'`
2. The same check should be added to the `"down"` listener above, in order to prevent setting `active` attribute when clicking link. Same applies to the `"keydown"` and `"keyup"` listeners.